apiserver/keymanager/keymanager_test.go

```diff
@@ -390,7 +475,28 @@ func (s *keyManagerSuite) TestImportKeys(c *gc.C) {
 			{Error: apiservertesting.ServerError(fmt.Sprintf("duplicate ssh key: %s", key2))},
 		},
 	})
-	s.assertEnvironKeys(c, append(initialKeys, key3, keymv[0], keymv[1], keymp[0], key4))
+	s.assertKeysForModel(c, st, append(initialKeys, key3, keymv[0], keymv[1], keymp[0], key4))
+}
+
+func (s *keyManagerSuite) TestImportKeys(c *gc.C) {
+	s.assertImportKeys(c, s.State, s.AdminUserTag(c), true)
+}
+
+func (s *keyManagerSuite) TestImportKeysSuperUser(c *gc.C) {
+	user := s.Factory.MakeUser(c, &factory.UserParams{Name: "superuser-fred", NoModelUser: true})
+	s.assertImportKeys(c, s.State, user.UserTag(), true)
+}
+
+func (s *keyManagerSuite) TestImportKeysModelAdmin(c *gc.C) {
+	otherModel := s.Factory.MakeModel(c, &factory.ModelParams{Name: "fred"})
+	defer otherModel.Close()
+	user := s.Factory.MakeUser(c, &factory.UserParams{Name: "admin" + otherModel.ModelTag().String()})
+	s.assertImportKeys(c, otherModel, user.UserTag(), true)
+}
+
```

Comment:
If it's even possible, please add a test for model owner that is not an admin. I don't know that it's sensible for the model to allow an owner to not be an admin, but the code caters for it so we should test it (esp. given the accidental ControllerTag use).